cli/flox-rust-sdk/src/providers/buildenv.rs

```diff
@@ -196,38 +479,683 @@ impl BuildEnv for BuildEnvNix {
+    fn flake_eval_failure() {
+        let locked_package = MockedLockedPackageFlake::builder()
+            .succeed_eval(false)
+            .unique(true)
+            .build();
+
+        let buildenv = BuildEnvNix;
+        assert!(
+            !buildenv
+                .check_store_path(locked_package.locked_installable.outputs.values())
+                .unwrap(),
+            "store path should be invalid before building"
+        );
+
+        let result = buildenv.realise_flakes(&locked_package);
+        let err = result.expect_err("realising flake should fail");
+        assert!(matches!(err, BuildEnvError::Realise2 { .. }));
+    }
+
+    /// Evaluation (and build) are skipped if the store path is already valid.
+    #[test]
+    fn flake_no_build_if_cached() {
+        let mut locked_package = MockedLockedPackageFlake::builder()
+            .succeed_eval(false)
+            .build();
+
+        for locked_path in locked_package.locked_installable.outputs.values_mut() {
+            *locked_path = env!("GIT_PKG").to_string();
+        }
+
+        let buildenv = BuildEnvNix;
+        assert!(
+            buildenv
+                .check_store_path(locked_package.locked_installable.outputs.values())
+                .unwrap(),
+            "store path should be valid before building"
+        );
+
+        let result = buildenv.realise_flakes(&locked_package);
+        assert!(result.is_ok(), "failed to skip building flake");
+    }
+}
+
+#[cfg(test)]
+mod realise_store_path_tests {
+    use super::*;
+    use crate::models::manifest::DEFAULT_PRIORITY;
+
+    fn mock_store_path(valid: bool) -> LockedPackageStorePath {
+        LockedPackageStorePath {
+            install_id: "mock".to_string(),
+            store_path: if valid {
+                env!("GIT_PKG").to_string()
+            } else {
+                "/nix/store/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx-invalid".to_string()
+            },
+            system: env!("NIX_TARGET_SYSTEM").to_string(),
+            priority: DEFAULT_PRIORITY,
+        }
+    }
+
+    #[test]
+    fn store_path_build_success_if_valid() {
+        let buildenv = BuildEnvNix;
+        let locked = mock_store_path(true);
+
+        // show that the store path is valid
+        assert!(buildenv.check_store_path([&locked.store_path]).unwrap());
+
+        buildenv
+            .realise_store_path(&locked)
+            .expect("an existing store path should realise");
+    }
+
+    #[test]
+    fn store_path_build_failure_if_invalid() {
+        let buildenv = BuildEnvNix;
+        let locked = mock_store_path(false);
+
+        // show that the store path is invalid
+        assert!(!buildenv.check_store_path([&locked.store_path]).unwrap());
+
+        let result = buildenv
+            .realise_store_path(&locked)
+            .expect_err("invalid store path should fail to realise");
+        assert!(matches!(result, BuildEnvError::Realise2 { .. }));
+    }
+}
+
+#[cfg(test)]
+mod buildenv_tests {
```

Comment:
These are a 1-to-1 implementation of the existing bats tests in `cli/tests/buildenv.bats`.
The latter depend explicitly on `pkgdb realise`.